pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/retention/RetentionManager.java

```diff
@@ -258,10 +258,11 @@ private void manageSegmentLineageCleanupForTable(TableConfig tableConfig) {
         if (SegmentLineageAccessHelper
             .writeSegmentLineage(_pinotHelixResourceManager.getPropertyStore(), segmentLineage, expectedVersion)) {
           // Delete segments based on the segment lineage
-          _pinotHelixResourceManager.deleteSegments(tableNameWithType, segmentsToDelete);
-          LOGGER.info("Finished cleaning up segment lineage for table: {}, deleted segments: {} in {}ms",
-              tableNameWithType, StringUtils.join(segmentsToDelete, ","),
-              (System.currentTimeMillis() - cleanupStartTime));
+          if (!segmentsToDelete.isEmpty()) {
```

Comment:
Should we also log a message to show the scenario when `segmentsToDelete` is empty?

Reply:
not really. We will have the following log anyway in case we successfully processed the table. If nothing is deleted, we don't need to add the log. (no log means no deletion happened)
```
    LOGGER.info("Segment lineage metadata clean-up is successfully processed for table: {}", tableNameWithType);
```